samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -357,4 +374,30 @@ class TaskInstance(
 
     startingOffset
   }
+
+  private def buildTaskContexts(epochTimeScheduler: EpochTimeScheduler):
```

Comment:
Minor: Would it be possible to refactor this into separate methods instead of a multi-value return? Prefer the former if it's simple to do, but otherwise this is fine.

Reply:
I agree that it would be better to avoid the multi-return value. However, I needed to save the taskContext and the applicationTaskContext, and I needed the taskContext to build the applicationTaskContext.
That said, I can just inline this back into the TaskInstance initialization section. It probably doesn't add too much now to move it into a separate method.